modules/services/cachix-agent.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cachix-agent;
+in {
+  options.services.cachix-agent = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable to run Cachix Agent as a system service.
+        
+        Read <link xlink:href="https://docs.cachix.org/deploy/">Cachix Deploy</link> documentation for more information.
+      '';
+    };
+
+    name = mkOption {
+      type = types.str;
+      default = config.networking.hostName;
+      description = ''
+        Agent name, usually the same as the hostname.
+      '';
+    };
+
+    package = mkOption {
+      description = ''
+        Package containing cachix executable.
+      '';
+      type = types.package;
+      default = pkgs.cachix;
+      defaultText = literalExample "pkgs.cachix";
+    };
+
+    credentialsFile = mkOption {
+      type = types.path;
+      default = "/etc/cachix-agent.token";
+      description = ''
+        Required file that needs to contain CACHIX_AGENT_TOKEN=...
+      '';
+    };
+
+    logFile = mkOption {
+      type = types.nullOr types.path;
+      default = "/var/log/cachix-agent.log";
+      description = "Absolute path to log all stderr and stdout";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    launchd.daemons.cachix-agent = {
```

Comment:
launchd daemons run as root by default, that's intended for this service?

Reply:
Yeah, it has to run as root.